modules/aleph/src/Aleph/Handler/AlephMaterialHandler.php

```diff
@@ -38,9 +38,10 @@ public function getHoldings(AlephMaterial $material) {
     $aleph_items = $this->client->getItems($material)->xpath('items/item');
     foreach ($aleph_items as $aleph_item) {
       $material = new AlephMaterial();
-      $items[] = $material::materialFromItem($aleph_item);
+      $items[(string) $aleph_item->xpath('z30-sub-library-code')[0]] =
+      $material::materialFromItem($aleph_item);
     }
-    return $items;
+    return array_filter($items, 'aleph_filter_items', ARRAY_FILTER_USE_KEY);
```

Comment:
It seems to be that we really want to filter on the value of `AlephMaterial->getSubLibraryCode()` code? If that is the case then there is no need to set the value as the key (duplicating the sublibrary extraction code) and use this as a filter subsequently.

You might as well do something like this:

```php
$items = array_map(function($aleph_item) {
	return AlephMaterial::materialFromItem($aleph_item);
}, $aleph_items);foreach ($aleph_items as $aleph_item) {
return array_filter($items, function(AlephMaterial $material) {
	return array_key_exists($material->getSubLibraryCode(), aleph_get_branches());
});
```

Note that I also removed `$material = new AlephMaterial();` - we do not have to create a new class instance to call our factory method as it is static. I also went all in on the functional programming :wink: